dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/ActivationWithLossLayer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2015 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn.layers;
+
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters.LayerIndex;
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters.NumberOfOutput;
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters.LossFunction;
+import edu.snu.dolphin.dnn.conf.NeuralNetworkConfigurationParameters.SerializedLayerConfiguartion;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.tang.formats.ConfigurationSerializer;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+/**
+ * Loss Layer with activation function.
```

Comment:
We should have comments saying that this layer is intended to be the final layer, in most cases.